htlcswitch/interceptable_switch.go

```diff
@@ -378,7 +383,10 @@ func (s *InterceptableSwitch) resolve(res *FwdResolution) error {
 
 	case FwdActionFail:
 		if len(res.FailureMessage) > 0 {
-			return intercepted.Fail(res.FailureMessage)
+			return intercepted.Fail(
+				res.FailureMessage,
+				!res.FailureMessageEncrypted,
```

Comment:
shouldn't this keep the same semantics? why change them here?

Reply:
I thought "encrypted" is the normal form, so wanted to use that as much as possible. But to keep the rpc backwards compatible, I needed `failuremessage_unencrypted` which defaults to false. Therefore the conversion.

Reply:
oh, didn't think about the backwards compat of the field. makes sense!